cudax/examples/stf/linear_algebra/cg_csr.cu

```diff
@@ -33,19 +36,22 @@ public:
   logical_data<slice<double>> val_handle;
   logical_data<slice<size_t>> row_handle;
   logical_data<slice<size_t>> col_handle;
+
+  context ctx;
 };
 
 class vector
 {
 public:
-  vector(size_t N)
+  vector(const context& _ctx, size_t N)
+      : ctx(_ctx)
   {
     handle = ctx.logical_data(shape_of<slice<double>>(N));
   }
 
   static void copy_vector(const vector& from, vector& to)
   {
-    ctx.parallel_for(to.handle.shape(), to.handle.write(), from.handle.read()).set_symbol("copy_vector")
+    to.ctx.parallel_for(to.handle.shape(), to.handle.write(), from.handle.read()).set_symbol("copy_vector")
```

Comment:
Under what circumstances could `to.ctx` be different from `from.ctx`?

Reply:
I should have stored a mutable context it was some constness issue

Reply:
oh ok